src/test/java/org/broadinstitute/consent/http/service/LibraryCardServiceTest.java

```diff
@@ -252,13 +272,28 @@ public void testUpdateLibraryCard_InvalidInstitution() {
     public void testDeleteLibraryCard_NotFound() {
         Institution institution = testInstitution();
         User user = testUser(institution.getId());
+        UserRole admin = new UserRole(UserRoles.ADMIN.getRoleId(), UserRoles.ADMIN.getRoleName());
+        user.addRole(admin);
         LibraryCard libraryCard = testLibraryCard(institution.getId(), user.getDacUserId());
         when(libraryCardDAO.findLibraryCardById(libraryCard.getId()))
                 .thenReturn(null);
         doNothing().when(libraryCardDAO).deleteLibraryCardById(any());
 
         initService();
-        service.deleteLibraryCardById(libraryCard.getId());
+        service.deleteLibraryCardById(libraryCard.getId(), user);
+    }
+
+    @Test(expected = ForbiddenException.class)
+    public void testDeleteLibraryCard_Forbidden_Error() {
+        Institution institution = testInstitution();
+        Integer modifiedId = institution.getId() + 1;
+        User user = testUser(institution.getId());
+        UserRole soRole = new UserRole(UserRoles.SIGNINGOFFICIAL.getRoleId(), UserRoles.SIGNINGOFFICIAL.getRoleName());
+        user.addRole(soRole);
+        LibraryCard libraryCard = testLibraryCard(modifiedId, user.getDacUserId());
+        when(libraryCardDAO.findLibraryCardById(any())).thenReturn(libraryCard);
+        initService();
+        service.deleteLibraryCardById(libraryCard.getId(), user);
     }
```

Comment:
Once this logic is in the resource class, this test should move to the resource class test.